src/main/php/predicate/ArraySubset.php

```diff
@@ -0,0 +1,90 @@
+<?php
+declare(strict_types=1);
+/**
+ * This file is part of bovigo\assert.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace bovigo\assert\predicate;
+
+use SebastianBergmann\Comparator\ComparisonFailure;
+use SebastianBergmann\Exporter\Exporter;
+
+class ArraySubset extends Predicate
+{
+    /**  @var array<mixed> */
+    private $other;
+    /** @var bool */
+    private $strict;
+    /** @var ComparisonFailure|null */
+    private $diff;
+
+    /**
+     * @param array<mixed> $other
+     */
+    public function __construct(array $other, bool $strict = false)
+    {
+        $this->other = $other;
+        $this->strict = $strict;
+    }
+
+    public function test($value): bool
+    {
```

Comment:
There should be a check if `$value` is indeed something iterable. Otherwise, a call to `toArray()` would lead to a `TypeError`. If `$value` is not iterable an \InvalidArgumentException should be thrown instead. See the `HasKey`or `Contains`predicates for similar checks.